core/gather/driver/target-manager.js

```diff
@@ -153,6 +171,38 @@ class TargetManager extends ProtocolEventEmitter {
     }
   }
 
+  /**
+   * @param {LH.Crdp.Runtime.ExecutionContextCreatedEvent} event
+   */
+  _onExecutionContextCreated(event) {
+    // This execution context was made via the protocol (e.g. by us or Puppeteer).
+    if (event.context.origin === '://' || event.context.origin === '') return;
+    // Just in case the above changes somehow.
+    if (event.context.name === '__puppeteer_utility_world__') return;
+    if (event.context.name === 'lighthouse_isolated_context') return;
+
+    const index = this._executionContextDescriptions.findIndex(d =>
+      d.uniqueId === event.context.uniqueId);
+    if (index === -1) {
```

Comment:
I'm not sure why, but puppeteer is sending a context created event twice for every single one. So I had to do a check on `uniqueId` here.